src/main/drivers/accgyro/accgyro_spi_icm426xx.c

```diff
@@ -103,11 +96,45 @@
 #define ICM426XX_INT_TPULSE_DURATION_100            (0 << ICM426XX_INT_TPULSE_DURATION_BIT)
 #define ICM426XX_INT_TPULSE_DURATION_8              (1 << ICM426XX_INT_TPULSE_DURATION_BIT)
 
-
 #define ICM426XX_RA_INT_SOURCE0                     0x65
 #define ICM426XX_UI_DRDY_INT1_EN_DISABLED           (0 << 3)
 #define ICM426XX_UI_DRDY_INT1_EN_ENABLED            (1 << 3)
 
+typedef struct odrConfig_s {
+    uint8_t odr;            // ODR: Output data rate (gyro sample rate in kHz)
+    uint8_t odr_idx;        // See GYRO_ODR in datasheet
+} odrConfig_t;
+
+typedef struct aafConfig_s {
```

Comment:
It may be better to store register values directly here (STATIC3 - STATIC5 / (STATIC2/2)-STATIC4), using macro to generate values

Reply:
You mean to just insert the macros here instead?

```c
#define ICM426XX_RA_GYRO_CONFIG_STATIC3             0x0C
#define ICM426XX_RA_GYRO_CONFIG_STATIC4             0x0D
#define ICM426XX_RA_GYRO_CONFIG_STATIC5             0x0E
#define ICM426XX_RA_ACCEL_CONFIG_STATIC2            0x03
#define ICM426XX_RA_ACCEL_CONFIG_STATIC3            0x04
#define ICM426XX_RA_ACCEL_CONFIG_STATIC4            0x05
```

or do you mean something like this?
```c
#define ICM426XX_RA_GYRO_AAF_CONFIG(x)      (0x0C + x)
#define ICM426XX_RA_ACCEL_AAF_CONFIG(x)     (0x03 + x)
```

or something completely different?

Reply:
No. The code writes 3 bytes into registers. So just store these bytes in aafConfig_s instead of values that need parsing.
And to keep input values identical, use macro to generate them from delt, deltSqr, shift

Reply:
Unfortunately there is a difference between the accelerometer and the gyro aaf configs: the accelerometer `delt` needs an additional bitshift (`<< 1`) in register ICM426XX_RA_ACCEL_CONFIG_STATIC2. Wouldn't that complicate things?

Reply:
Only a little bit. If is is always *2, hardcode it for now. If it makes sense to use different values, use 2 delt values in aafConfig_s, and even better, store all 6 values, so that gyro and acc can be decoupled in future ..

Reply:
Sorry, I cannot imagine what you want me to do here which doesn't increase the complexity and decreases the readability of the aaf config table. Also gyro and acc configs are already decoupled right now. So I'm not sure if I understand you correctly.

Reply:
Sorry, missed gyro/acc difference. See my comment, but that is just suggestion